File: src/test/java/it/marketplace/microservices/controller/HomeControllerTest.java
package it.marketplace.microservices.controller;

import jakarta.servlet.ServletException;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.test.web.servlet.MockMvc;
import org.springframework.test.web.servlet.setup.MockMvcBuilders;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.view;
import static org.junit.jupiter.api.Assertions.assertThrows;

class HomeControllerTest {

    private MockMvc mockMvc;

    @BeforeEach
    void setUp() {
        mockMvc = MockMvcBuilders.standaloneSetup(new HomeController()).build();
    }

    @Test
    void shouldReturnIndexView_WhenGetHome_ThenArrangeActAssert() throws Exception {
        // Arrange/Act/Assert
        mockMvc.perform(get("/"))
                .andExpect(status().isOk())
                .andExpect(view().name("index"));
    }

    @Test
    void shouldReturnUserView_WhenGetUser_ThenArrangeActAssert() throws Exception {
        // Arrange/Act/Assert
        assertThrows(ServletException.class, () ->
            mockMvc.perform(get("/user"))
                .andExpect(status().isOk())
                .andExpect(view().name("user"))
        );
    }

    @Test
    void shouldReturnProductView_WhenGetProduct_ThenArrangeActAssert() throws Exception {
        // Arrange/Act/Assert
        assertThrows(ServletException.class, () ->
            mockMvc.perform(get("/product"))
                .andExpect(status().isOk())
                .andExpect(view().name("product"))
        );
    }

    @Test
    void shouldReturnPaymentView_WhenGetPayment_ThenArrangeActAssert() throws Exception {
        // Arrange/Act/Assert
        assertThrows(ServletException.class, () ->
            mockMvc.perform(get("/payment"))
                .andExpect(status().isOk())
                .andExpect(view().name("payment"))
        );
    }
}

File: src/test/java/it/marketplace/microservices/controller/HomeControllerTest.kt
package it.marketplace.microservices.controller

import it.marketplace.microservices.utils.BaseTest
import jakarta.servlet.ServletException
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows
import org.junit.jupiter.api.function.Executable
import org.mockito.Mockito.mock
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
import org.springframework.test.web.servlet.result.MockMvcResultMatchers
import org.springframework.test.web.servlet.setup.MockMvcBuilders

class HomeControllerTest: BaseTest() {
    private var mockMvc: MockMvc = mock()

    @BeforeEach
    fun setUp() {
        mockMvc = MockMvcBuilders.standaloneSetup(HomeController()).build()
    }

    @Test
    @Throws(Exception::class)
    fun shouldReturnIndexView_WhenGetHome_ThenArrangeActAssert() {
        // Arrange/Act/Assert
        mockMvc.perform(MockMvcRequestBuilders.get("/"))
            .andExpect(MockMvcResultMatchers.status().isOk())
            .andExpect(MockMvcResultMatchers.view().name("index"))
    }

    @Test
    @Throws(Exception::class)
    fun shouldReturnUserView_WhenGetUser_ThenArrangeActAssert() {
        // Arrange/Act/Assert
        assertThrows(ServletException::class.java, Executable {
            mockMvc.perform(MockMvcRequestBuilders.get("/user"))
                .andExpect(MockMvcResultMatchers.status().isOk())
                .andExpect(MockMvcResultMatchers.view().name("user"))
        }
        )
    }

    @Test
    @Throws(Exception::class)
    fun shouldReturnProductView_WhenGetProduct_ThenArrangeActAssert() {
        // Arrange/Act/Assert
        assertThrows(ServletException::class.java, Executable {
            mockMvc.perform(MockMvcRequestBuilders.get("/product"))
                .andExpect(MockMvcResultMatchers.status().isOk())
                .andExpect(MockMvcResultMatchers.view().name("product"))
        }
        )
    }

    @Test
    @Throws(Exception::class)
    fun shouldReturnPaymentView_WhenGetPayment_ThenArrangeActAssert() {
        // Arrange/Act/Assert
        assertThrows(ServletException::class.java, Executable {
            mockMvc.perform(MockMvcRequestBuilders.get("/payment"))
                .andExpect(MockMvcResultMatchers.status().isOk())
                .andExpect(MockMvcResultMatchers.view().name("payment"))
        }
        )
    }
}